docs: mention wrong URLs as a cause of git authentication errors (#15304)
rust-lang/cargo: src/doc/src/appendix/git-authentication.md

This can be enabled with the `CARGO_NET_GIT_FETCH_WITH_CLI=true` environment
variable.

> > **Note:** Cargo does not require authentication for public git dependencies
> > so if you see an authentication failure in that context, ensure that the
> > URL is correct.

Reviewer: Thanks for the pull request!

I think the problem is more like libgit2 doesn't provide a prompt and ask for credentials. Cargo may still require credentials if a Git repo is specified with SSH protocol. The sentences here are great but a bit imprecise. Not sure how to rephrase it. Maybe something like this?

```suggestion
> **Note:** Cargo does not provide a prompt for Git authentication.
> If you see an authentication failure, first ensure that the URL is correct.
```

Author: Hmm, I feel like that leaves out the specific case I was thinking about, which is an intentionally public repo not needing authentication and thus not benefiting from a prompt. I usually use SSH, but intentionally put a `git = 'https://github.com/xiphoseer/fontdue.git'` URL there for it to work in CI without extra steps.

Reviewer: I think yeah my suggestion did lose the  most important point of your patch. This note doesn't need to be 100% perfect reflecting every technical detail.

Thanks for the help. Going to merge this now!
